File: calculaPreco.ts
function calcularPrecoFinal(produto) {
  const desconto = produto.desconto ?? 0;

  const precoFinal = produto.preco - (produto.preco * desconto / 100);

  // Linha nova criada a partir da cópia da linha do preço final:
  console.log(`Preço inicial de ${produto.nome}: R$ ${produto.preco.toFixed(2)}`);

  // Linha original (copiada e mantida):
  console.log(`Preço final de ${produto.nome}: R$ ${precoFinal.toFixed(2)}`);
}

// Exemplo de uso
const produto1 = {
  nome: "Fone de ouvido",
  preco: 200,
  desconto: 25 // alterado de 15 para 25
};

calcularPrecoFinal(produto1);

File: calculaPreco.py
# Define função para calcular preço final
def calcular_preco_final(nome, preco, desconto=0):
    preco_final = preco - (preco * desconto / 100)

    # Linha adicionada: preço inicial
    print(f"Preço inicial de {nome}: R$ {preco:.2f}")

    # Linha original (copiada e mantida)
    print(f"Preço final de {nome}: R$ {preco_final:.2f}")


# Exemplo de uso com desconto alterado para 25%
calcular_preco_final("Fone de ouvido", 200, 25)
